draft-ietf-tls-dtls13.md

```diff
@@ -347,6 +392,52 @@ allowing the sequence number to wrap.
 Implementations MUST NOT allow the epoch to wrap, but instead MUST establish
 a new association, terminating the old association.
 
+### Determining the Header Format
+
+Implementations can distinguish the two header formats by examining
+the first byte, which in the DTLSCiphertext header represents the
+content type. The only valid values for the content type are
+alert(21), handshake(22), application_data(23), and ack(25),
+with application_data actually
+representing protected data with the true content type being
+encrypted. If any of these values is present, then the record MUST be
+handled as DTLSCiphertext.
+
+If the first byte is any other other value, then receivers
+MUST check to see if the leading bits of the first byte are
+001. If so, they MUST process the record as DTLSShortCiphertext.
+Otherwise, the record MUST be rejected as if it had failed
+deprotection.
+
+### Reconstructing the Sequence Number and Epoch {#reconstructing}
+
+When receiving a DTLSShortCiphertext message, the recipient does not
+have a full epoch or sequence number value and so there is some
+opportunity for ambiguity.  Because the full epoch and sequence number
+are used to compute the per-record nonce, failure to reconstruct these
+values leads to failure to deprotect the record, and so implementations
+MAY use a mechanism of their choice to determine the full values.
+This section provides an algorithm which is comparatively simple
+and which implementations are RECOMMENDED to follow.
+
+If the epoch bit has the same parity as the current epoch, then
+implementations SHOULD reconstruct the sequence number by computing
+the full sequence number which is numerically closest to the most
+recently successfully deprotected record.
```

Comment:
Missed this:  it isn't "most recently".  In QUIC we use the numeric closest to the highest successfully decrypted packet plus one.  Or the closest to the next expected packet.  Otherwise delayed packets will shift the window backwards.  (Basically, this means that negative differences are used to allow for delays, and positive differences for lost packets).